src/http_provides.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2024 Canonical Ltd.
```

Comment:
For comparison, here is the reactive provider

https://github.com/juju-solutions/interface-http/blob/master/provides.py